src/cloudai/workloads/ai_dynamo/ai_dynamo.sh

```diff
@@ -93,22 +167,122 @@ _patch_dynamo_args() {
   fi
 
   dynamo_args["url"]="http://${dynamo_args["frontend-node"]}:${dynamo_args["port"]}"
+
+  _validate_or_build_nodelists
 }
 
-_patch_section_args() {
-  prefill_args["--model"]=${dynamo_args["model"]}
-  decode_args["--model"]=${dynamo_args["model"]}
+_gpus_per_node() {
+  local n=$(echo "${CUDA_VISIBLE_DEVICES:-}" | tr ',' '\n' | grep -c . || true)
+  [[ "$n" -gt 0 ]] && echo "$n" || echo "1"
+}
+
+_resolve_host_ip() {
+  local host="$1"
+  local ip
+  ip="$(getent ahosts "$host" | grep STREAM | head -n1 | awk '{print $1}')"
+  if [[ -z "$ip" ]]; then
+    log "ERROR: Could not resolve IP for host $host"
+    exit 1
+  fi
+  echo "$ip"
+}
+
+_apply_sglang_section_args() {
+  prefill_args["--port"]=${dynamo_args["prefill-port"]}
+  decode_args["--port"]=${dynamo_args["decode-port"]}
```

Comment:
Nit: I used some patterns like `%port%` and `%model%` in toml that get replaced with dynamo_args["port"] and dynamo_args["model"]. This might scale a bit better when dealing with multiple backends.